app/models/chargeHistory/ChargeHistoryModel.scala

```diff
@@ -31,14 +30,12 @@ case class ChargeHistoryModel(taxYear: String,
                               poaAdjustmentReason: Option[String]) {
 
   val reasonCode: String = {
-    if (poaAdjustmentReason.isDefined) "adjustment"
+    if (poaAdjustmentReason.isDefined) Adjustment.asString
     else
       reversalReason match {
-        case "amended return" => "amend"
-        case "Customer Request" => "request"
-        case error =>
-          Logger("application").error(s"Missing or non-matching history reason: $error found")
-          "unrecognisedReason"
+        case "amended return" => Amend.asString
+        case "Customer Request" => Request.asString
+        case _ => "unrecognisedReason"
```

Comment:
This doesn't translate to a message key... The type signature for this method should be Option[String] where the case _ scenario would be None. Then wherever this is used we should do some validation, and if the value of this val is None then we should do the logging and maybe return an exception somewhere since this should not happen.